scripts/zkai_native_block_boundary_pivot_selector_gate.py

```
@@ -0,0 +1,696 @@
+    expect_equal(require_int(summary.get("gkr_smallest_width_preserving_bytes"), "GKR width-preserving bytes"), GKR_SMALLEST_WIDTH_PRESERVING_BYTES, "GKR width-preserving bytes")
+    expect_equal(require_int(summary.get("mlp_fused_typed_bytes"), "MLP fused typed"), MLP_FUSED_TYPED_BYTES, "MLP fused typed")
+    expect_equal(require_int(summary.get("mlp_fusion_typed_saving_bytes"), "MLP saving"), MLP_FUSION_TYPED_SAVING_BYTES, "MLP saving")
+    expect_equal(summary.get("mlp_fusion_typed_saving_ratio"), MLP_FUSION_TYPED_SAVING_RATIO, "MLP saving ratio")
+    expect_equal(require_int(summary.get("compact_preprocessed_typed_bytes"), "compact preprocessed typed"), COMPACT_PREPROCESSED_TYPED_BYTES, "compact preprocessed typed")
+    expect_equal(summary.get("compact_preprocessed_vs_nanozk_ratio"), COMPACT_PREPROCESSED_VS_NANOZK_RATIO, "compact preprocessed ratio")
+
+    routes = require_list(payload.get("routes"), "routes")
+    expect_equal([require_str(row.get("route_id"), "route id") for row in routes], list(EXPECTED_ROUTE_IDS), "route order")
+    for row in routes:
+        if set(row) != set(ROW_COLUMNS) | {"non_claims"}:
+            raise PivotSelectorError("route row key inventory drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)